### PR TITLE
restore the "natural order" without renaming percentiles to positions & this time without conflicts

### PR DESCRIPTION
**Pull request by tomtomme**
Apparently I cannot change a name of a function  like "percentileValues" without breaking everything, so I reverted that and also gave the conflicts another try.

The really new changes I made are:
~~a) Changed section and variable name percentileValues to postions (as this better reflects the content of the section. Is this ok, or do other modules depend on the exact variable name?~~
b) Every textbook I know begins with mode median mean in this order. JASP reflects this now also in the output table.
c) Sum is a better fit for section distribution

If this commit is accepted I will reorder further items in this module to make it more beginner friendly.

### CODE REVIEW

File: R/descriptives.R [hunk 1]
```diff
@@ -273,9 +273,8 @@ Descriptives <- function(jaspResults, dataset, options) {
   if (numberMissingSplitBy) 
     stats$addFootnote(message=gettextf("Excluded %1$i rows from the analysis that correspond to the missing values of the split-by variable %2$s", numberMissingSplitBy, options$splitby))
   
-
-  stats$dependOn(c("splitby", "variables", "percentileValuesEqualGroupsNo", "percentileValuesPercentilesPercentiles", "mean", "standardErrorMean",
-    "median", "mode", "standardDeviation", "cOfVariation", "variance", "skewness", "kurtosis", "shapiro", "range", "iqr", "mad","madrobust", "minimum", "maximum",
+  stats$dependOn(c("splitby", "variables", "percentileValuesEqualGroupsNo", "percentileValuesPercentilesPercentiles", "mode", "median", "mean", "standardErrorMean",
+    "standardDeviation", "cOfVariation", "variance", "skewness", "kurtosis", "shapiro", "range", "iqr", "mad", "madrobust", "minimum", "maximum",
```
Comment (vandenman) [inline R/descriptives.R:277]: The order here does not matter, so I'd revert these changes

File: R/descriptives.R [hunk 2]
```diff
@@ -406,15 +406,15 @@ Descriptives <- function(jaspResults, dataset, options) {
   resultsCol[["Valid"]]   <- length(na.omitted)
   resultsCol[["Missing"]] <- rows - length(na.omitted)
 
-  if (base::is.factor(na.omitted) && (options$mean || options$mode || options$median || options$minimum || options$standardErrorMean || options$iqr || options$mad || options$madrobust || options$kurtosis || options$shapiro || options$skewness || options$percentileValuesQuartiles || options$variance || options$standardDeviation ||  options$cOfVariation || options$percentileValuesPercentiles || options$sum || options$maximum)) {
+  if (base::is.factor(na.omitted) && (options$mode || options$median || options$mean || options$minimum || options$standardErrorMean || options$iqr || options$mad || options$madrobust || options$kurtosis || options$shapiro || options$skewness || options$percentileValuesQuartiles || options$variance || options$standardDeviation ||  options$cOfVariation || options$percentileValuesPercentiles || options$sum || options$maximum)) {
```
Comment (vandenman) [inline R/descriptives.R:409]: The order here does not matter, so I'd revert these changes

File: R/descriptives.R [hunk 3]
```diff
@@ -406,15 +406,15 @@ Descriptives <- function(jaspResults, dataset, options) {
...
     shouldAddNominalTextFootnote <- TRUE
   }
   
   shouldAddIdenticalFootnote <- all(na.omitted[1] == na.omitted) && (options$skewness || options$kurtosis || options$shapiro)  
   
+  resultsCol[["Median"]]                  <- .descriptivesDescriptivesTable_subFunction_OptionChecker(options$median,            na.omitted, median)
```
Comment (vandenman) [inline R/descriptives.R:415]: The order here does not matter, so I'd revert these changes

File: R/descriptives.R [hunk 4]
```diff
@@ -289,10 +288,10 @@ Descriptives <- function(jaspResults, dataset, options) {
   stats$addColumnInfo(name="Valid",   title=gettext("Valid"), type="integer")
   stats$addColumnInfo(name="Missing", title=gettext("Missing"), type="integer")
 
+  if (options$mode)                 stats$addColumnInfo(name="Mode",                        title=gettext("Mode"),                    type="number")
+  if (options$median)               stats$addColumnInfo(name="Median",                      title=gettext("Median"),                  type="number")
   if (options$mean)                 stats$addColumnInfo(name="Mean",                        title=gettext("Mean"), 				            type="number")
```
Comment (vandenman) [inline R/descriptives.R:293]: The order here is the only thing that matters, so I think these two lines (and the `if (options$sum) ...`) are the only required changes.

**Review by vandenman — CHANGES_REQUESTED** [review]
So the changes mostly look fine. Some changes don't actually change anything, so it would be great if you could revert those (I left a comment in the code wherever that is the case).

However, I'm not 100% sure about two changes to the interface. In particular, `Positions` might not be an optimal name, and `Sum` doesn't really belong to the category `Distribution`.

Old interface:

image

New Interface:
image

@EJWagenmakers, @TimKDJ any comments regarding `Positions`, `Sum`, or something else about the new UI?

**Comment by tomtomme** [comment]
Thanks for the review, I will look into it.

Regarding the Wording "Percentile values" (old) "Positions" (new): The idea was, that the old name was not really correct.  quartiles, quintiles, centiles, percentiles together are called quantiles. So an alternative would be "Quantile values". But that seems too long and to complicated. Postions ist what quantiles are. But just "Quantiles" without "values" would do too. We also do not have "Disperstion values" or "central tendency values".

Regarding Sum: It does not really belong anywhere. It was part of the central tendency group which makes no sense to me. In the group "Distribution" at least it makes some sense, because it sums up the whole distribution.

Cheers, Thomas

**Comment by AlexanderLyNL** [comment]
I like the suggestion "Quantiles" it's nice and snappy and more accurate than what was used before.

File: R/descriptives.R (hunk 1 above)
Comment (tomtomme) [reply R/descriptives.R:277]: I thought its easier to read and track changes, if the order is the same in QT and R. But you can decide.

**Comment by tomtomme** [comment]
So I switched Positions to Quantiles. That should be all.

**Review by vandenman — APPROVED** [review]
